Add Deploy on Koyeb guide (#1508)
meilisearch/documentation: learn/cookbooks/koyeb.md

# Deploy Meilisearch on Koyeb

## Introduction

> This guide explains how to deploy a ready-to-use Meilisearch service on Koyeb.

Reviewer: ```suggestion
This guide explains how to deploy a ready-to-use Meilisearch instance on Koyeb.
```

[Koyeb](https://www.koyeb.com) is a developer-friendly serverless platform to deploy apps globally. The platform lets you seamlessly run Docker containers, web apps, and APIs with git-based deployment, TLS encryption, native autoscaling, a global edge network, and built-in service mesh & discovery.

## Requirements

To successfully follow and complete this guide, you need a [Koyeb account](https://app.koyeb.com).

## Deploy Meilisearch

> The fastest way to deploy Meilisearch on Koyeb is to use the **Deploy to Koyeb** button to deploy. It allows you to have your Meilisearch instance running with minimal configuration.

Reviewer: ```suggestion
The fastest way to deploy Meilisearch on Koyeb is to use the **Deploy to Koyeb** button.
```

Take care to replace the `MEILI_MASTER_KEY` environment variable value with a strong key to secure your Meilisearch instance.

You can for instance run the following command from the terminal to generate a strong secret to use as a master key:

```bash
python -c 'import os,base64; print(base64.urlsafe_b64encode(os.urandom(32)).decode())'
```

::: warning
Setting up a master key is optional but strongly recommended when running in production. Unsetting the master key will make your Meilisearch instance unprotected and publicly accessible.
:::

> ### Test Meilisearch.

Reviewer: ```suggestion
### Test Meilisearch
```

> Copy the public URL (e.g. `https://meili-myorg.koyeb.app`) from the [Koyeb control panel](https://app.koyeb.com) and paste it your browser.

Reviewer: ```suggestion
Copy the public URL (e.g. `https://meili-myorg.koyeb.app`) from the [Koyeb control panel](https://app.koyeb.com) and paste it in your browser.
```

> You land on the MeiliSearch dashboard, where you are asked to authenticate using your master key.

Reviewer: ```suggestion
You should land on the Meilisearch [search preview](/learn/what_is_meilisearch/search_preview.md), where you are asked to enter your master key.
```

> You are now ready to create your first index!

Reviewer: ```suggestion
You are now ready to [create your first index](/learn/getting_started/quick_start.md)!
```
